test/e2e_flags.go

```diff
@@ -58,22 +61,49 @@ type ServingEnvironmentFlags struct {
 	Https            bool // Indicates where the test service will be created with https
 }
 
-// initializeServingFlags registers flags used by e2e tests, calling flag.Parse() here would fail in
-// go1.13+, see https://github.com/knative/test-infra/issues/1329 for details
 func initializeServingFlags() *ServingEnvironmentFlags {
 	var f ServingEnvironmentFlags
 
+	// Only define flags here. Attempting flag.Set() will silently fail.
 	flag.BoolVar(&f.ResolvableDomain, "resolvabledomain", false,
 		"Set this flag to true if you have configured the `domainSuffix` on your Route controller to a domain that will resolve to your test cluster.")
 	flag.BoolVar(&f.Https, "https", false,
 		"Set this flag to true to run all tests with https.")
 
-	flag.Set("alsologtostderr", "true")
-	logging.InitializeLogger(test.Flags.LogVerbose)
-
-	if test.Flags.EmitMetrics {
-		logging.InitializeMetricExporter(E2EMetricExporter)
-	}
+	klog.InitFlags(nil)
 
 	return &f
 }
+
+var flagsSetup = false
+var flagsMutex = &sync.Mutex{}
+
+func SetupServingFlags() {
+	flagsMutex.Lock()
+	if !flagsSetup {
+
+		flag.Set("alsologtostderr", "true")
+
+		if test.Flags.LogVerbose {
+			// If klog verbosity is not set to a non-default value (via "-args -v=X"),
+			// set up verbosity for klog so round_trippers.go prints:
+			//   URL, request headers, response headers, and response body
+			// Definition in vendor/k8s.io/client-go/transport/round_trippers.go:DebugWrappers
+			if flag.CommandLine.Lookup("v").Value.String() == "0" {
+				flag.Set("v", "8")
+			}
+			fmt.Printf("Test Flags: {")
```

Comment:
Do we need this section?
It is probably needed for you to debug, but not beyond?

Reply:
All the systems I have worked on have printed out test flags and environment variables when they run. I thought it would would be valuable to keep (and, as apparently the sole user, I feel my opinion is the best :-P)

Reply:
Why only print this when `test.Flags.LogVerbose`? Also this should be pulled out into a `printFlags` functions.

Reply:
I don't want to clutter logs if people aren't asking for verbosity. I will pull to printFlags function.

Reply:
I thought the generated test executable only would get called once, and thus this function would only print once, but for some reason it seems to run almost every time except when jobs are parallelized, then only one arbitrary one runs it.

Reply:
(by "almost every time" I mean "for every test function")